connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -106,13 +107,14 @@ message Unknown {}
 
 // Common metadata of all relations.
 message RelationCommon {
-  // TODO(SPARK-48639): Add origin like Expression.ExpressionCommon
-
-  // (Required) Shared relation metadata.
-  string source_info = 1;
+  // (Optional) Shared relation metadata.
+  string source_info = 1 [deprecated=true];
```

Comment:
Seems `source_info` is unused yet in the codebase ... ?

Reply:
why do we add this field at the first place?

Reply:
I think we wanted to leverage it somewhere codepoint but I think we ended up with not using it. @grundprinzip

Reply:
I wanted to do what origin does. You can simply remove this field and mark the id reserved.
```suggestion
  reserved 1;
```

Reply:
This actually complains that it's a breaking change:

```
Run bufbuild/buf-breaking-action@v1
Error: Previously present field "1" with name "source_info" on message "RelationCommon" was deleted.
Error: buf found 1 breaking changes.
```

I can port this change to branch-3.5 if that's fine.